lib/winapi/profiling.c

```diff
@@ -3,22 +3,59 @@
 // SPDX-FileCopyrightText: 2019 Stefan Schmidt
 
 #include <profileapi.h>
+#ifdef USE_RDTSC_FOR_FREQ
+#include <synchapi.h>
+#endif
 #include <assert.h>
 #include <stddef.h>
 #include <xboxkrnl/xboxkrnl.h>
 
+#ifdef USE_RDTSC_FOR_FREQ
+static LARGE_INTEGER frequency = {{0, 0}};
+static void __attribute__((constructor)) PrimeQueryPerformanceFrequency ()
+{
+    ULARGE_INTEGER f_rdtsc = {{0, 0}}, s_rdtsc = {{0, 0}};
+    ULONG f_ticks = 0, s_ticks = 0;
+
+    KeEnterCriticalRegion();
+
+    // The values generated after launching aren't accurate, give it time to increment...
+    Sleep(700);
+
+    f_rdtsc.QuadPart = __rdtsc();
+    f_ticks = KeTickCount;
+
+    Sleep(200);
+
+    s_rdtsc.QuadPart = __rdtsc();
+    s_ticks = KeTickCount;
+
+    s_rdtsc.QuadPart -= f_rdtsc.QuadPart;
+    s_rdtsc.QuadPart /= s_ticks - f_ticks;
```

Comment:
dumb question, but what are `s_` and `f_` for? Why not `start`/`end` or `before`/`after` or `begin`/`finish` or `a`/`b` or ...?

Reply:
Seems like a real big nit and probably shouldn't have been commented. I'll change it if it's bothering. It's private variables in a function that isn't meant to be called by a developer.

Reply:
Addressed in [8ea50ce](https://github.com/XboxDev/nxdk/pull/663/commits/8ea50ce3ddfd8697c9f5c190df4cf7e0dae90c06)

Reply:
>  It's private variables in a function that isn't meant to be called by a developer.

❓ ❗ ❓ 

It's public code in an open-source project that needs to be maintained and should be easily readable and extensible.
It should also follow the established code-convention (of surrounding code).

Reply:
`f_` first, we can tell this by looking at the placement when setting the variables value. `s_` second. It's been addressed. Not sure the need for the additional comment.